andreeapetcu/assignment3/TestWordSearch.java

```diff
@@ -0,0 +1,407 @@
+import java.util.*;
+
+public class TestWordSearch {
+
+  private WordSearch wordSearch;
+  private char[][] validGrid;
+  private TrieNode[][] validTrieGrid;
+  private int numberRowsGrid;
+  private int numberColumnsGrid;
+  private Dictionary validDict;
+  private List<String> validListOfWords;
+  private char[][] nullGrid;
+  private Dictionary nullDict;
+  private Dictionary emptyDict;
+  private List<String> emptyListOfDictionaryWords;
+  private boolean[][] validVisited;
+  private boolean[][] nullVisited;
+  private ArrayList<String> validWordsAccumulator;
+  private ArrayList<String> nullWordsAccumulator;
+  private TrieNode validTrieNode;
+  private TrieNode nullTrieNode;
+  private TrieNode[][] nullTrieGrid;
+
+
+  /**
+   * Checks if a list contains a given trie node.
+   *
+   * @param list the list to be searched
+   * @param node the node of interest
+   * @return true if the node is in the list, false otherwise
+   */
+  private boolean containsNode (LinkedList<TrieNode> list, TrieNode node) {
```

Comment:
This doesn't need to be a LinkedList, so I'd just declare it a list in the method prototype.

(and as an aside - it is almost never true that the Java LinkedList class is the right solution to a problem. There is nearly always a better solution.)